app/controllers/publish/providers_controller.rb

```diff
@@ -101,7 +101,7 @@ def search
     private
 
     def provider
-      @provider ||= recruitment_cycle.providers.find_by!(provider_code: params[:provider_code] || params[:code])
+      @provider ||= recruitment_cycle.providers.find_by(provider_code: params[:provider_code] || params[:code])
```

Comment:
This seems like an important change.

Is the intention here to raise a 403 Forbidden error instead of a 404 Not Found error when a provider is not found within a recruitment cycle?

Reply:
We call `provider` in a before action now, whereas we didn't before.
Some actions do not require a provider, `/publisher/organisations` for example.
There is not provider parameter in this case and no Provider can be found.
That will trigger a 404.

I haven't thought enough about how to do this better.

Reply:
🤔